docs: add detail on spot interruption behavior (#3789)
aws/karpenter: website/content/en/preview/concepts/deprovisioning.md

When Karpenter detects one of these events will occur to your nodes, it automatically cordons, drains, and terminates the node(s) ahead of the interruption event to give the maximum amount of time for workload cleanup prior to compute disruption. This enables scenarios where the `terminationGracePeriod` for your workloads may be long or cleanup for your workloads is critical, and you want enough time to be able to gracefully clean-up your pods.

> For Spot interruptions, the provisioner will start a new instance as soon as it sees the Spot interruption warning. Spot interruptions have a __2 minute notice__ before Amazon EC2 reclaims the instance. Karpenter's average node startup time means that generally, this is sufficient time for the new node to become ready and to move the pods to the new node. Karpenter will respect the PDB while it drains the node. This means there is a chance that a node may not be fully drained by the time it is reclaimed.

Reviewer: ```suggestion
For Spot interruptions, the provisioner will start a new machine as soon as it sees the Spot interruption warning. Spot interruptions have a __2 minute notice__ before Amazon EC2 reclaims the instance. Karpenter's average node startup time means that, generally, there is sufficient time for the new node to become ready and to move the pods to the new node before the machine is reclaimed.
```

Small updates

Author: cool, sounds good - just updated

  ...
```

> For Spot Interruption Warnings, Karpenter will cordon and drain the node as per the above. There may be insufficient capacity to schedule the pods running on the affected node. The provisioner will then start a new instance to schedule these pods. Spot Interruptions have a 2 minute notice before Amazon EC2 reclaims the instance. Generally, this is sufficient time for the new node to become ready and to schedule the pods. Karpenter will follow the PDB while it drains the node. This means there is a chance that a node may not be fully drained by the time it is reclaimed.

Reviewer: Can you add this detail higher up where we say "When Karpenter detects..." at the end. I'm imagining the key details that we need to include here beyond what's currently there is that:
1. Karpenter spins up a new instance as soon as it sees the spot interruption
2. The spot interruption comes with a 2-minute warning, which is effectively the time that Karpenter has to spin up the new node
3. Mention that Karpenter's average node startup time means that this 2 minute startup is generally enough to spin up the new node and transfer the pods over.

Author: Thanks @jonathan-innis  ! Addressed in latest commit
